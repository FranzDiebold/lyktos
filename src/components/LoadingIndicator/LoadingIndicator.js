import React from 'react';

import './LoadingIndicator.scss';

function LoadingIndicator() {
    return (
        <div className="has-text-centered">
            <div className="loading-spinner"></div>
        </div>
    );
}

export default LoadingIndicator;
