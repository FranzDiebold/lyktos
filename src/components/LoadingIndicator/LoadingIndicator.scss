$color: #25b0cd;
$size: 62px;

.loading-spinner {
    margin: 6rem 0;
    display: inline-block;
    width: $size;
    height: $size;

    &:after {
        content: " ";
        display: block;
        width: $size;
        height: $size;
        margin: 1px;
        border-radius: 50%;
        border: 5px solid $color;
        border-color: $color transparent $color transparent;
        animation: loading-spinner 1.2s linear infinite;
    }

    @keyframes loading-spinner {
        0% {
            transform: rotate(0deg);
        }

        100% {
            transform: rotate(360deg);
        }
    }
}
  