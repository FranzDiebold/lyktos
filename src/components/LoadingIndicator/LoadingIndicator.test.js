import React from 'react';
import renderer from 'react-test-renderer';

import LoadingIndicator from './LoadingIndicator';

it('renders loading indicator correctly', () => {
    const tree = renderer
        .create(<LoadingIndicator />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
