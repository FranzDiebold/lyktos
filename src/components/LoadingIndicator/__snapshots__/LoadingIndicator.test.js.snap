// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading indicator correctly 1`] = `
<div
  className="has-text-centered"
>
  <div
    className="loading-spinner"
  />
</div>
`;
